features/mib-compiler/src/main/java/org/opennms/features/mibcompiler/services/JsmiMibParser.java

```diff
@@ -451,6 +451,8 @@ private String getMetricType(SmiPrimitiveType type) {
             return "string";
         if (type.equals(SmiPrimitiveType.UNSIGNED_32)) // Unsigned32 will be treated as integer.
             return "integer";
+        if (type.equals(SmiPrimitiveType.IP_ADDRESS)) // IpAddress will be treated as strings.
+            return "string";
         return type.toString().replaceAll("_", "").toLowerCase();
```

Comment:
Will this ever return a correct type, or should we just default to "string" here instead?

Reply:
From my conversation with @jeffgdotorg  IpAddress type should be treated as string when generating collection files. I believe only those values allowed in the datacollection-config.xsd file should remain. 
The XML parser will fail to parse a type is not allowed in the datacollection-config.xsd and Jeff mentioned something like this type shouldn't be allowed because is some sort of custom type... I think.